src/Features/Core/Portable/ValidateFormatString/AbstractValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,295 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.LanguageServices;
+
+namespace Microsoft.CodeAnalysis.ValidateFormatString
+{
+    internal abstract class AbstractValidateFormatStringDiagnosticAnalyzer<TSyntaxKind> :
+        DiagnosticAnalyzer
```

Comment:
@CyrusNajmabadi But it's *not* a code style analyzer. There's nothing code style related about "your string.Format call is going to crash". Sure, there might be convenience, but there's other *meaning* behind being a code style analyzer that just doesn't fit here (most code style analyzers have options, this doesn't, etc.). I think it should be left alone. Thoughts?